pkg/kubectl/cmd/util/factory.go

```diff
@@ -940,16 +940,44 @@ func GetFirstPod(client coreclient.PodsGetter, namespace string, selector labels
 	return pod, 1, nil
 }
 
-// TODO: We need to filter out stuff like secrets.
-func (f *factory) Command() string {
+func (f *factory) Command(enableSecret bool) string {
 	if len(os.Args) == 0 {
 		return ""
 	}
 	base := filepath.Base(os.Args[0])
 	args := append([]string{base}, os.Args[1:]...)
+
+	for i := 0; i < len(args); i++ {
+		if strings.HasPrefix(args[i], "--") {
+			name := (args[i])[2:]
+			if len(name) == 0 || name[0] == '-' || name[0] == '=' {
+				return ""
+			}
+
+			split := strings.SplitN(name, "=", 2)
+			name = split[0]
+
+			flag := f.flags.Lookup(name)
+			if flag != nil && flag.Secret {
```

Comment:
I do not think that we need `flag.Secret` / changes to pflag. Instead set an annotation on the flag back when it was created and check for that annotation here.